scripts/gen_ref_nav.py

```diff
@@ -0,0 +1,47 @@
+# Adapted from https://github.com/mkdocstrings/python/blob/main/scripts/gen_ref_nav.py
+# License: ISC License - Copyright (c) 2021, Timothée Mazzucotelli
+
+"""Generate the code reference pages and navigation.
+
+No need to run this script manually, it is called by mkdocs-material during the build process.
+
+You can run it manually to test the output.
+"""
+
+from pathlib import Path
+
+import mkdocs_gen_files
+
+nav = mkdocs_gen_files.Nav()
+mod_symbol = '<code class="doc-symbol doc-symbol-nav doc-symbol-module"></code>'
+
+root = Path(__file__).parent.parent
+src = root / "src"
+
+for path in sorted(src.rglob("*.py")):
```

Comment:
Down the line, my preference would be to document only the public API:
- `src/django_components/__init__.py`
- `src/django_components/templatetags/component_tags.py`
- `src/django_components/middleware.py`